### DISCUSSION TITLE
v24.12.10

### DISCUSSION THREAD

**Discussion by davidrapan** [davidrapan/ha-solarman#313] [Announcements]
## What's Changed
* [refactor: Read/Write w/ asyncio.Semaphore](https://github.com/davidrapan/ha-solarman/commit/c54750920cfbf29da1c0ba2b958e8813626fcb79)
  * Removed **Wait for attempts** argument from service actions

**Full Changelog**: https://github.com/davidrapan/ha-solarman/compare/v24.12.08...v24.12.10

<hr /><em>This discussion was created from the release <a href='https://github.com/davidrapan/ha-solarman/releases/tag/v24.12.10'>v24.12.10</a>.</em>

**Comment by sofkaski** [comment]
I installed the release in my environment (Deye SUN-8K-SG04P3-EU with a logger with firmware version LSW3_15_FFFF_1.0.9E.) and based on some poking around (entities are updated from the inverter and changes propagate to the inverter) the release seems to be working.

<details>
<summary> Note to people who might do manual updates ... </summary>
`pysolarmanv5` is nowadays included as a submodule to `ha-solarman` repository.
If you are using a ZIP-release noting special to note since the module is included in the zip. 
But, if you are using `ha-solarman` repository clone remember to init and update submodules before  copying the content from the clone to homeassistant config. There are likely several ways to do this, but I do it this way:

- I have a clone of my `ha-solarman` fork that I keep in sync with upstream from @davidrapan. (Lately I have not done any changes of my own, but the fork is there if I want to do some contribution)
-  I am running my homeassistant in a Raspberry PI with `HAOS`.
- I have cloned my fork to my home assistant environment `/share/ha-solarman`

When I do an update of `ha-solarman` I do the following in a shell in `HAOS` (started with `ssh <my username>@homeassistant.local` from my development machine).

1. `cd /share/ha-solarman`
1. `git fetch --tags origin` 
2. `git checkout <release tag (or any commit of interest)>`
3. `git submodule init && git submodule update`  <-- this is the new thing to remember
5. ` cp -r custom_components/solarman/* /homeassistant/custom_components/solarman/`
6. `ha core rebuild`
7. Tadaa...! Home assistant should be running after a while   

</details>

**Comment by ebagnoli** [comment]
I have lost some parameters lile Load_L1, names are changing too often, and automations and all the rest needs to be changed every moment... Why are all these name changement necessary?

**Reply by davidrapan** [reply]
There were no changes w/ naming but filtered out non-relevant sensors.

**Reply by ebagnoli** [reply]
Sorry, probably it was not a change, but I had to change all my parameters all around HA, anyway thanks for your excellent work, even if today it doesn't load anymore. I think is a problem in the core but I don't hnow how to solve it.
Regards

**Reply by davidrapan** [reply]
What about the latest version? 😉

**Reply by sofkaski** [reply]
> What about the latest version? 😉

I installed `v24.12.10_3` to my environment and did some silk glove testing. The release seems to be working. So maybe the myth of Friday and 13th day of the month is busted.
